### ISSUE TITLE
Bug in example for DiagGGN Second order optimizer

### ISSUE THREAD

**Issue by AlexImmer** [f-dangel/backpack#95]
Thanks for this great library!

I found a bug in the example showcasing the second-order implementations for writing an optimizer (https://docs.backpack.pt/en/master/use_cases/example_diag_ggn_optimizer.html#sphx-glr-use-cases-example-diag-ggn-optimizer-py).
This mistake is within the example in the documentation and might hinder people to get started with using approximate second-order methods.

```python
losses = []
accuracies = []
for batch_idx, (x, y) in enumerate(mnist_loader):
    x, y = x.to(DEVICE), y.to(DEVICE)
    outputs = model(x)
    loss = loss_function(outputs, y)
```

is missing the zero-ing of gradients. While backpack seems to overwrite iteratively on a `.backward()`, the gradient just accumulates here. I ran the example for several epochs which results in a divergence because the gradient is never reset, I don't think this is intended right?

```python
losses = []
accuracies = []
for batch_idx, (x, y) in enumerate(mnist_loader):
    x, y = x.to(DEVICE), y.to(DEVICE)
    # ---------
    model.zero_grad()
    # ---------
    outputs = model(x)
    loss = loss_function(outputs, y)
```
This changes the example quite drastically and the learning rate probably needs to be adjusted to get the same learning curve..

Hope this helps.

**Comment by f-dangel** [comment]
Hi Alex, thanks for reporting that one!

We're going to fix the example.

**Comment by fKunstner** [comment]
👍 Thanks for the catch! [Fixed version is up](https://docs.backpack.pt/en/master/use_cases/example_diag_ggn_optimizer.html#sphx-glr-use-cases-example-diag-ggn-optimizer-py)
